hilti/toolchain/src/config.cc.in

```diff
@@ -10,6 +12,50 @@ using namespace hilti;
 const auto flatten = util::flattenParts;
 const auto prefix = util::prefixParts;
 
+namespace {
+std::optional<hilti::rt::filesystem::path> precompiled_libhilti(const Configuration& configuration, bool debug) {
+    // We disable use of precompiled headers for sanitizers builds since the
+    // sanitizer flags are not exposed on the config level.
+    //
+    // TODO(bbannier): Allow using of precompiled headers for sanitizer builds.
+#ifdef HILTI_HAVE_SANITIZER
+    return {};
+#endif
+
+    // If we are executing from the build directory, the cache is also located
+    // there; else it lives in a versioned folder in the user's `$HOME/.cache/spicy`.
+    std::optional<rt::filesystem::path> cache;
+    if ( configuration.uses_build_directory ) {
+        cache = configuration.build_directory / "include" / "hilti" / "rt" / "autogen";
+    }
+    else {
+        const char* homedir;
+
+        if ( (homedir = getenv("HOME")) == NULL ) {
+            auto pwuid = getpwuid(getuid());
+            if ( ! pwuid )
+                return {};
+
+            homedir = pwuid->pw_dir;
+
+            if ( homedir )
+                cache = rt::filesystem::path(rt::filesystem::path(homedir) / ".cache" / "spicy" /
+                                             configuration.version_string);
+        }
+    }
+
+    if ( ! cache )
+        return {};
```

Comment:
This looks something that could be more generally useful. I suggest factoring that out into `util::cacheDirectory().` 

And maybe relocate the build-dir version it from `<build>/include/hilti/rt/autogen` to `<build>/cache`, so that it matches the structure we use for the user/system?